browser/brave_browser_process_impl.cc

```diff
@@ -128,7 +128,7 @@ BraveBrowserProcessImpl::BraveBrowserProcessImpl(StartupData* startup_data)
   // Create P3A Service early to catch more histograms. The full initialization
   // should be started once browser process impl is ready.
   brave_p3a_service();
-  brave::SetupHistogramsBraveization();
+  histogram_braveizer_ = brave::HistogramsBraveizer::Create();
```

Comment:
thanks @mkarolin and @mariospr , the approach looks good. The braveizer probably doesn't need to be refcounted - I'll try to fix